src/tensor/tensor.h

```diff
@@ -6,6 +6,28 @@ namespace kaldi {
 namespace tensor {
 
 
+// Similar to llvm/PyTorch's ArrayRef, this is a lightweight way to store an
+// array (zero or more elements of type T).  The array is not owned here; it
+// will generally be unsafe to use an ArrayRef as other than a local variable.
+template <typename T>
+struct ArrayRef {
+  // Note:
+  uint64_t size;
+  int64_t *data;
+
+  // Lots to do here.
+
+  inline int64_t operator (uint64_t i) const {
+    KALDI_ASSERT(i < size);
+    return data[i];
+  }
+
+  // cast to std::vector; for cases where you might need to
+  // change the contents or keep it more permanently.
+  operator std::vector<int64_t> () const;
+};
+
+
 enum {
```

Comment:
Can we avoid unnamed unscoped enums? These were problematic to wrap with CLIF. I am guessing they will also be problematic when wrapping with pybind11.

Reply:
What is the preferred alternative, `enum class` or the "good old enum" declared within a class scope? I use `enum class` once in a while, I like the better scoping the offer.